Implement Meetup GraphQL compatibility

irby opened the pull request:
Add in GraphQL support for the Meetup API

bogdankharchenko commented on app-modules/event-importer/src/Providers/Keys/.gitignore:
```diff
@@ -0,0 +1 @@
+*.pem
```
Let's bring this up to the parent .gitignore and delete this "Keys" directory all together.  Since we wont commit any keys, I don't think we need this folder.

allella replied:
Just curious since this is a file, and we probably aren't sharing it with all future devs, will running the import break if someone doesn't have a .pem file. Or, are we able to gracefully fail in that case?

irby replied:
It will break if the runner doesn't have a pem file and at least one org has `meetup_graphql` set. It will not fail gracefully, and not sure if we'd want it to (what happens if the server tries to run this import and it doesn't have the key?)

allella replied:
The concern is more for devs. 

They obviously won't have the production .pem and I'm wondering if we foresee devs wanting to run the event import and if the Meetup GraphQL import without a .pem is going to make that entire process fail, or if it would be fairly easy to avoid or workaround.

irby replied:
That's fair. I think we could use an environment variable like APP_ENV where devs would run it with `dev` and print out a warning for devs but throw an exception for prod. I'll defer to @bogdankharchenko on the approach we should take

bogdankharchenko commented on config/event-import-handlers.php:
```diff
@@ -3,19 +3,26 @@
 return [
     'max_days_in_past' => env('EVENT_IMPORTER_MAX_DAYS_IN_PAST', 30),
     'max_days_in_future' => env('EVENT_IMPORTER_MAX_DAYS_IN_FUTURE', 180),
+    'meetup_graphql_private_key_path' => env('EVENT_IMPORTER_MEETUP_GRAPHQL_PRIVATE_KEY_PATH', __DIR__ . '/../app-modules/event-importer/src/Providers/Keys/meetup-graphql.pem'),
```
```suggestion
    'meetup_graphql_private_key_path' => env('EVENT_IMPORTER_MEETUP_GRAPHQL_PRIVATE_KEY_PATH'),
```

We should just leave this blank -- if the pem file missing, let's throw an exception in the handler.

bogdankharchenko commented on app-modules/event-importer/src/Providers/MeetupGraphqlTokenProvider.php:
```diff
@@ -0,0 +1,37 @@
+
+    protected function getJwtKey()
+    {
+        $privateKey = file_get_contents(config('event-import-handlers.meetup_graphql_private_key_path'));
```
Before we perform file_get_contents -- we should, 
1. Check if the env is actually set -- if it is not set, throw a runtime exception
2. Ensure file exists.

irby replied:
Sure, makes sense. Will push a change out later to fix this.

bogdankharchenko reviewed (changes requested):
@irby great work on the import!  A few small house keeping things around where the code lives, but overall it works on my machine =)

irby commented on app-modules/event-importer/tests/Feature/MeetupRestTest.php:
```diff
@@ -41,7 +41,6 @@ public function test_active_meetup_event_is_imported_correctly(): void
         $this->assertEquals(1, $active_event->rsvp_count);
         $this->assertEquals(1702944000, $active_event->active_at->utc()->unix());
         $this->assertEquals('https://www.meetup.com/code-for-the-carolinas-greenville/events/lvgwftyfcqbxb/', $active_event->uri);
-        $this->assertNull($active_event->cancelled_at);
```
Nah, I just felt that since we're checking on the cancelled_at property being set in one test, it makes sense to assert that cancelled_at is not set under certain conditions.

irby commented on app-modules/event-importer/src/Services/MeetupGraphqlHandler.php:
```diff
@@ -0,0 +1,233 @@
+    private function filterEvents(array $events): array
+    {
+        $start_date = now()->subDays($this->max_days_in_past)->startOfDay();
+        $end_date = now()->addDays($this->max_days_in_future)->startOfDay();
```
Made change.

irby commented:
@bogdankharchenko I have addressed all feedback -- one outstanding question from @allella regarding how to handle local development with respect to the required private key file.

https://github.com/hackgvl/hackgreenville-com/pull/345#discussion_r1693041347

bogdankharchenko commented:
Overall this looks good - one more small comment about tests, but we can live with the exceptions being thrown for missing api keys, missing pem files -- we currently throw exeptions for everbrite as well. 

We may need to introduce a generic exception later which we can catch more efficiently

irby commented on app-modules/event-importer/tests/Feature/MeetupGraphqlTest.php:
```diff
@@ -0,0 +1,215 @@
+        return file_get_contents(__DIR__ . '/../fixtures/meetup-graphql/' . $file);
+    }
+
+    private function setupTest(): void
```
Changed!
